runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java

```diff
@@ -137,6 +139,50 @@ protected void shutDown() throws Exception
 		cannonPosition = null;
 		cballsLeft = 0;
 		removeCounter();
+		skipProjectileCheckThisTick = false;
+		spotPoints.clear();
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(ItemContainerChanged event)
+	{
+		if (event.getItemContainer() != client.getItemContainer(InventoryID.INVENTORY))
+		{
+			return;
+		}
+
+		int partsFound = 0;
+
+		for (Item item : event.getItemContainer().getItems())
+		{
+			if (item == null)
+			{
+				continue;
+			}
+
+			switch (item.getId())
+			{
+				case ItemID.CANNON_BASE:
+				case ItemID.CANNON_STAND:
+				case ItemID.CANNON_BARRELS:
+				case ItemID.CANNON_FURNACE:
+					partsFound++;
```

Comment:
May want to use a Set instead since this can be fooled by duplicate items. Not sure if its worth the effort though.